include/zephyr/bluetooth/conn.h

```diff
@@ -718,6 +718,20 @@ void bt_conn_foreach(enum bt_conn_type type,
  */
 struct bt_conn *bt_conn_lookup_addr_le(uint8_t id, const bt_addr_le_t *peer);
 
+/**
+ * @brief Look up an existing connection by index.
+ *
+ *  Look up an existing connection based on the connection index.
+ *
+ *  The caller gets a new reference to the connection object which must be
+ *  released with bt_conn_unref() once done using the object.
+ *
+ * @param index Connection index from @ref bt_conn_index
+ *
+ *  @return Connection object or NULL if not found.
+ */
+struct bt_conn *bt_conn_lookup_index(uint8_t index);
```

Comment:
From https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#defining-smaller-prs

> If introducing a new API, the PR must include an example usage of the API. This provides context to the reviewer and prevents submitting PRs with unused APIs.

Is there any sample or test that will use this? Given that it's an inverse of an existing API, it may not be needed

We should also consider adding this to the release notes as "new API function"

Reply:
Added to the release notes, the tests and samples would be the existing usage of the function in the host I suppose.
The function already exists, this is just adding docs and making it public.